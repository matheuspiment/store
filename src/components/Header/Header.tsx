import React from "react";
import { isInteger } from "lodash";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import Badge from "@material-ui/core/Badge";
import ShoppingCart from "@material-ui/icons/ShoppingCart";

const useStyles = makeStyles(() => ({
  grow: {
    flexGrow: 1,
  },
  title: {
    display: "block",
  },
  sectionDesktop: {
    display: "flex",
  },
}));

const Header = ({
  itemsAmount = 0,
  onClickOnCart
}: {
  itemsAmount?: number;
  onClickOnCart: () => void;
}) => {
  const classes = useStyles();

  const shoppingCartLabel =
    isInteger(itemsAmount) && itemsAmount > 0
      ? `show ${itemsAmount} items in the cart`
      : "No items in your cart";

  return (
    <div className={classes.grow}>
      <AppBar position="static">
        <Toolbar>
          <Typography className={classes.title} variant="h6" noWrap>
            Store
          </Typography>
          <div className={classes.grow} />
          <div className={classes.sectionDesktop}>
            <IconButton data-testid="btn-cart" aria-label={shoppingCartLabel} color="inherit" onClick={onClickOnCart}>
              <Badge badgeContent={itemsAmount} color="secondary">
                <ShoppingCart />
              </Badge>
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
