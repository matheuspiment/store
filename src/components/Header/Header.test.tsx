import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import Header from './Header';

describe('Header', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <Header itemsAmount={1} onClickOnCart={jest.fn()}/>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should trigger onClickOnCart', () => {
    const handleClickOnCart = jest.fn();

    const { getByTestId } = render(
      <Header itemsAmount={1} onClickOnCart={handleClickOnCart}/>,
    );

    fireEvent.click(getByTestId('btn-cart'));

    expect(handleClickOnCart).toHaveBeenCalledTimes(1);
  });
})


