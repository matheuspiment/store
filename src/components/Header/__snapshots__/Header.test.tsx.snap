// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="makeStyles-grow-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h6 MuiTypography-noWrap"
      >
        Store
      </h6>
      <div
        className="makeStyles-grow-1"
      />
      <div
        className="makeStyles-sectionDesktop-3"
      >
        <button
          aria-label="show 1 items in the cart"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="btn-cart"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
              >
                1
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </header>
</div>
`;
